.github/workflows/tests.yml

```diff
@@ -0,0 +1,42 @@
+name: "integration-tests"
+on: # rebuild any PRs and main branch changes
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - provider/**
+
+env:
+  PULUMI_MOUNT_DIR: /home/vscode/.temp_pulumi
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout (GitHub)
+        uses: actions/checkout@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create mount directories for the AWS and Pulumai
+        run: mkdir ~/.aws
```

Comment:
Are you sure we need that step ?
AFAIK `pulumi login` does create the .pulumi folder, and the same should be true for `aws configure` (?)

Reply:
It's kinda confusing that we have two different handlings for AWS & Pulumi

Shouldn't we create dirs with `mkdir -p ${PATH}` ?
This way if the directory exists (mounted from a local devcontainer) then it's fine, if it doesn't then AWS & Pulumi have directories to store their settings into

I still find it weird that we need this at all : I'm pretty sure that on a new computer, you don't have a ~/.aws folder yet the `aws configure` command executes just fine

Reply:
What if we did the `pulumi login` and the `aws configure` steps in the GH workflow before running the devcontainer task ?

That's what a normal developer would do right ? Set these credentials locally on their machine, then run a container to start working on a project